linode_api4/objects/serializable.py

```diff
@@ -117,7 +151,55 @@ def _serialize(self) -> Dict[str, Any]:
         """
         Serializes this object into a JSON dict.
         """
-        return asdict(self)
+        cls = type(self)
+        type_hints = get_type_hints(cls)
+
+        def attempt_serialize(value: Any) -> Any:
+            """
+            Attempts to serialize the given value, else returns the value unchanged.
+            """
+            if issubclass(type(value), JSONObject):
+                return value._serialize()
+
+            return value
+
+        def should_include(key: str, value: Any) -> bool:
+            """
+            Returns whether the given key/value pair should be included in the resulting dict.
+            """
+
+            if key in cls.always_include:
+                return True
+
+            hint = type_hints.get(key)
+
+            # We want to exclude any Optional values that are None
+            # NOTE: We need to check for Union here because Optional is an alias of Union.
+            if (
+                hint is None
+                or get_origin(hint) is not Union
+                or type(None) not in get_args(hint)
+            ):
+                return True
+
+            return value is not None
+
+        result = {}
+
+        for k, v in vars(self).items():
+            if not should_include(k, v):
+                continue
+
+            if isinstance(v, List):
+                v = [attempt_serialize(j) for j in v]
+            elif isinstance(v, Dict):
+                v = {k: attempt_serialize(j) for k, j in v.items()}
+            else:
+                v = attempt_serialize(v)
+
+            result[k] = v
+
+        return result
```

Comment:
Unfortunately we have to reinvent the wheel here because we don't have access to type hints or `always_include` for nested JSONObjects in the asdict `dict_factory` method.

Ideally I'd want to break this out into a separate PR but excluding optional fields from generated request bodies is a requirement for `LKEClusterControlPlaneACL`.